opentracing-api-extensions-tracer-spring-autoconfigure/src/main/java/io/opentracing/contrib/api/tracer/spring/autoconfigure/TracerBeanPostProcessor.java

```diff
@@ -38,21 +39,34 @@
     public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
         if (tracerObservers != null && bean instanceof Tracer) {
             boolean observerFound = false;
-            APIExtensionsTracer tracer = new APIExtensionsTracer((Tracer)bean);
+            APIExtensionsManager tracer = getManager((Tracer)bean);
             for (TracerObserver observer : tracerObservers) {
                 if (observer != null) {
                     observerFound = true;
                     tracer.addTracerObserver(observer);
                 }
             }
             if (observerFound) {
-                log.info("Use extensions API tracer to wrap tracer=" + bean + " with observers=" + tracerObservers);
+                log.info("Initialized extensions API manager/tracer="
+                        + tracer + " with observers=" + tracerObservers);
                 return tracer;
             }
         }
         return bean;
     }
 
+    static APIExtensionsManager getManager(Tracer tracer) {
+        APIExtensionsManager manager = null;
+        if (tracer instanceof APIExtensionsManager) {
+            manager = (APIExtensionsManager)tracer;
+            log.info("Use existing extensions API manager/tracer=" + manager);
+        } else {
+            manager = new APIExtensionsTracer(tracer);
+            log.info("Use extensions API wrapper tracer=" + manager + " to wrap original tracer=" + tracer);
+        }
```

Comment:
Personally I wouldn't use `log.info` here, but rather `debug` or even lower.
There is already an info message in the main method `"Initialized extensions API...."`.

However, as this is only used during Spring initialization, I don't feel strongly about it, just my opinion.